### ISSUE TITLE
new neural architectures [suggestion]

### ISSUE THREAD

**Issue by mrocr** [lquirosd/P2PaLA#22]
@lquirosd Thank you for your hard work
This topic was created only to bring your attention at the new advancements in neural architectures relating to semantic segmentation.

The goal is to find a neural network that is capable of delivering Resnet-50's accuracy, without being over-parameterized or heavy. Thus, the idea of finding that secret sauce, that is able to run with speed & accuracy even when used on a CPU **is actually possible, and have been already achieved.**

Have a look at:
- **M2U-Net**
0.55M parameters, can achieve +96% rate, it takes 577ms to 1.67s when running on CPU.
https://arxiv.org/abs/1811.07738
https://github.com/laibe/M2U-Net

- **NAS-Unet**
0.8M parameters, can achieve +98% rate.
https://ieeexplore.ieee.org/stamp/stamp.jsp?tp=&arnumber=8681706
https://github.com/tianbaochou/NasUnet

- **ESFNet**
0.18 parameters, 85.34 IoU, 2.513 flops, 100 to 143 fps.
https://arxiv.org/abs/1903.12337
https://github.com/mrluin/ESFNet-Pytorch

**Comment by lquirosd** [comment]
Thanks for the comments, those a re very important updates that can be added to P2PaLA.
Please feel free to pull-request any update you made on that direction.

**Comment by mrocr** [comment]
Thank you for your hard work.
